src/include/platform/CHIPDeviceEvent.h

```diff
@@ -255,6 +255,11 @@ enum PublicEventTypes
      * Signals that secure session is established.
      */
     kSecureSessionEstablished,
+
+    /**
+     * Signals that socket select operation to be started.
+     */
+    kSocketSelectStart,
```

Comment:
This and the struct could be ok platform specific events rather than `PublicEventTypes`

Reply:
As discussed this will be a generic event.